lib/config/loader.js

```diff
@@ -196,6 +196,53 @@ module.exports = function (userCfgPath) {
 					default: userConfig.login && userConfig.login.twitch && userConfig.login.twitch.enabled ? null : []
 				}
 			},
+			discord: {
+				enabled: {
+					doc: 'Whether to enable Discord authentication.',
+					format: Boolean,
+					default: false
+				},
+				clientID: {
+					doc: 'A Discord application ClientID https://discord.com/developers/applications/',
+					format: String,
+
+					// This will throw if the user does not provide a value, but only if Discord auth is enabled.
+					default: userConfig.login && userConfig.login.discord && userConfig.login.discord.enabled ? null : ''
+				},
+				clientSecret: {
+					doc: 'A Discord application ClientSecret https://discord.com/developers/applications/',
+					format: String,
+
+					// This will throw if the user does not provide a value, but only if Discord auth is enabled.
+					default: userConfig.login && userConfig.login.discord && userConfig.login.discord.enabled ? null : ''
+				},
+				scope: {
+					doc: 'A space-separated string of Discord application permissions.',
+					format: String,
+					default: 'identify'
+				},
+				allowedIDs: {
+					doc: 'Which Discord IDs to allow',
+					format: VALIDATE_STRING_ARRAY,
+
+					// This will throw if the user does not provide a value and is not using requiredGuildIDs, but only if Discord auth is enabled.
+					default: userConfig.login && userConfig.login.discord && userConfig.login.discord.enabled && !userConfig.login.discord.allowedGuildIDs ? null : []
+				},
+				allowedGuildIDs: {
+					doc: 'Users in these Discord Guilds are allowed to login',
+					format: VALIDATE_STRING_ARRAY,
+
+					// This will throw if the user does not provide a value and is not using allowedIDs, but only if Discord auth is enabled.
+					default: userConfig.login && userConfig.login.discord && userConfig.login.discord.enabled && !userConfig.login.discord.allowedIDs ? null : []
+				},
+				guildRequiredPermissions: {
+					doc: 'In addition to being in one of allowedGuildIDs, the user must have these permissions in the guild.',
+					format: VALIDATE_STRING_ARRAY,
+
+					// By default, the user must not have any permissions in the guild
+					default: []
+				}
```

Comment:
I love this idea of using guilds. A couple of things I'd like to note:
- It should be array of objects that has a guild ID and permissions, so if you can have guilds with different permission setups.
- Using role object IDs instead of guild permissions would allow more flexibility and makes more sense to me.

Reply:
I added this in the newest commit